cmd/clusterctl/pkg/client/config/reader_viper.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog"
+)
+
+// viperReader implements Reader using viper as backend for reading from environment variables
+// and from a clusterctl config file.
+type viperReader struct {
```

Comment:
yikes, viper support should be dropped from clusterctl ideally.
it's a unmaintained library..

Reply:
Is it unmaintained? It looks to have recent commits and just published a v1.6.0 release an hour ago.

Reply:
it definitely was unmaintained 1/2 year ago, maybe the maintainers are back on track.
WG component standard did not really agree on using Viper k8s-wide for solving the "flags vs config" problem, but there are exceptions like the test/e2e of k/k - which is more of a tech dept at this point.

Reply:
the config reader is pluggable, so we can change it any time